src/bsk201909/secrets.md

```diff
@@ -0,0 +1,264 @@
+
+It's important to ensure that those secrets are stored as securely as possible.
+
+Let's recap on how secrets work in Kubernetes.
+
+Secrets are objects that contain key-value pairs, and some metadata.
+
+Secrets are similar to ConfigMaps and share the same limitations (1MB in size — as an example).
+
+The main differences are the security protections added by Kubernetes teams to secrets.
+
+This is how a secret looks like in a YAML representation:
+
+```yaml|title=secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: MWYyZDFlMmU2N2Rm
+```
+
+You can create a secret like any other Kubernetes object (usually with kubectl).
+
+The manifest will be sent to the server that will proceed it and store it in etcd - similarly to all Kubernetes objects.
+
+_So why did the Kubernetes team decided to “encrypt” the secrets using base64?_
+
+As you might already know, [base64 is an encoding, not an encryption](https://en.wikipedia.org/wiki/Base64).
+
+Encoding allows you to represent binary data in a manifest.
+
+Imagine storing a certificate in Kubernetes without base64: a lot of "���" — unrecognised characters.
+
+Base64 translates those binaries files in standard strings such as "aGVsbG8gdGhlcmUh==".
+
+But we still haven't answered how to properly secure those secrets.
+
+In Kubernetes, you can opt-in to encryption at rest: by enabling this feature, Kubernetes API encrypts the secrets (optionally, using an external KMS system) before storing them in etcd.
+
+You solved the issue of storing sensitive files such as certificates inside the cluster.
+
+You also protected your secrets at rest with a suitable encryption provider.
+
+_Is your cluster \_finally_ secure?\_
+
+**Perhaps.**
+
+However, the secrets that you load into the cluster must exist somewhere.
+
+_Do you keep a copy or rely on Kubernetes to be the only source of truth?_
+
+_How do you back them up?_
+
+_What if you keep a copy and they go out of sync?_
+
+You could store the secrets with the other manifests files - for example, in Git.
+
+That could solve most of the challenges related to secret management:
+
+- You get a full audit history for free thanks to GIT
+- You can reuse the same merging strategy and approve changes to your secrets as you do with the rest of the code
+- Your code and your secrets are kept in sync at all times
+
+_But can you secure the secrets in GIT?_
+
+_Can anyone who has access to the repository run away with your precious credentials?_
+
+There are some existing tools that let you create "encrypted secrets" that can be stored on GIT alongside the rest of the deployment files.
+
+The tools also provide a mechanism to decrypt back to regular secrets so your app can consume them seamlessly.
+
+A popular project in this space is Sealed Secrets.
+
+Sealed secrets has two parts: an operator deployed into your cluster and a command-line tool designed to interact with it called `kubeseal`.
+
```

Comment:
Explain how to install the client tool:

`````markdown
You can install the command-line tool with:

```terminal|command=1|title=bash
brew install kubeseal
```
`````

Maybe for Linux too?